validator/das/das.go

```diff
@@ -0,0 +1,175 @@
+)
+
+func Init() {
+	if validator.PlatformFactories == nil {
+		validator.PlatformFactories = make(map[types.Platform]func(*validator.Base) validator.IValidator)
+	}
+	validator.PlatformFactories[types.Platforms.Das] = func(base *validator.Base) validator.IValidator {
+		das := Das{base}
+		return &das
+	}
+}
+
+func (das *Das) GeneratePostPayload() (post map[string]string) {
+	das.Identity = strings.ToLower(das.Identity)
+	payload := DasSignPayload{
+		Version:     "1",
+		Comment:     "Here's an NextID proof of this DAS account.",
+		Comment2:    "To validate, base64.decode the signature, and recover pubkey from it using sign_payload with ethereum personal_sign algo.",
+		Persona:     "0x" + mycrypto.CompressedPubkeyHex(das.Pubkey),
+		BitAddress:  das.Identity,
+		SignPayload: das.GenerateSignPayload(),
+		Signature:   "%SIG_BASE64%",
+		CreatedAt:   util.TimeToTimestampString(das.CreatedAt),
+		Uuid:        das.Uuid.String(),
+	}
+	payload_json, _ := json.MarshalIndent(payload, "", "\t")
+	return map[string]string{"default": string(payload_json)}
+}
+
+func (das *Das) GenerateSignPayload() (payload string) {
+	das.Identity = strings.ToLower(das.Identity)
+	payloadStruct := validator.H{
+		"action":     string(das.Action),
+		"identity":   das.Identity,
+		"platform":   string(types.Platforms.Das),
+		"prev":       nil,
+		"created_at": util.TimeToTimestampString(das.CreatedAt),
+		"uuid":       das.Uuid.String(),
+	}
+	if das.Previous != "" {
+		payloadStruct["prev"] = das.Previous
+	}
+
+	payloadBytes, err := json.Marshal(payloadStruct)
+	if err != nil {
+		l.Warnf("Error when marshaling struct: %s", err.Error())
+		return ""
+	}
+
+	return string(payloadBytes)
+}
+
+func (das *Das) Validate() (err error) {
+	das.Identity = strings.ToLower(das.Identity)
+	das.SignaturePayload = das.GenerateSignPayload()
+
+	das.ProofLocation = fmt.Sprintf(LocationURL, das.Identity)
+	req, err := json.Marshal(DasRequest{Account: das.Identity})
+	if err != nil {
+		return xerrors.Errorf("Error when marshalling request: %w", err)
+	}
+
+	resp, err := http.Post(URL, "application/json", bytes.NewReader(req))
+	if err != nil {
+		return xerrors.Errorf("Error when requesting proof: %s", err.Error())
+	}
+	if resp.StatusCode != 200 {
+		return xerrors.Errorf("Error when requesting proof: Status code %d", resp.StatusCode)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return xerrors.Errorf("Error when getting resp body")
+	}
+
+	result := new(DasResponse)
+	err = json.Unmarshal(body, result)
+	if err != nil {
+		return xerrors.Errorf("error when decoding JSON: %w", err)
+	}
+
+	return das.validateRecord(result)
+}
+
+func (das *Das) validateRecord(resp *DasResponse) error {
+	if resp.ErrorNumber != 0 {
+		return xerrors.Errorf("err_no %d: %s", resp.ErrorNumber, resp.ErrorMessage)
+	}
+	keyName := KeyPrefix + "0x" + mycrypto.CompressedPubkeyHex(das.Pubkey)
+	value := ""
+	for _, r := range resp.Data.Records {
```

Comment:
now we have `lo.ForEach()`

Find helper func here to refactor this part of code:

https://github.com/samber/lo